migrations/20240112041245_create-calculate-folder-size-function.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const functionName = 'calculate_folder_size';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  up: async (queryInterface) => {
+    await queryInterface.sequelize.query(`
+        CREATE OR REPLACE FUNCTION ${functionName}(folder_uuid_param UUID) RETURNS BIGINT AS $$
```

Comment:
Here are the advantages of using this function instead of a recursive query:

Clarity and Maintainability:
Procedural code can be easier to understand and maintain for some developers, especially those familiar with procedural programming languages like PL/pgSQL.
The use of control flow statements (such as loops and conditionals) can make the code clearer compared to recursive SQL queries, which might be harder to grasp for some.

Performance:
In some cases, stored functions can be more efficient in terms of performance than recursive SQL queries, particularly when dealing with complex operations or large datasets.
The stored function may perform specific optimizations or access indexes more effectively.

Flexibility:
Using PL/pgSQL provides flexibility to perform more advanced operations within the function, such as handling exceptions, executing custom business logic, etc.


My conclusion: 

I understand well the benefits that you mentioned above, but taking into account this use case that we only need the size of the files that belong to each folder, we need to have more freedom, by freedom I mean only taking the properties that we are going to need, make the smallest number of queries and joins possible and target the fields that have indexes, that freedom is offered to us by PLSQL and not CTE.

In this file, I leave you the queries in both PSQL and CTE.

I would like you to backup the production database to test the queries and see which one is better.

[queries.txt](https://github.com/internxt/drive-server-wip/files/13930048/queries.txt)

Reply:
great, you're completly right!